ext/js/data/sandbox/anki-note-data-creator.js

```diff
@@ -170,6 +170,65 @@ function getPublicContext(context) {
     };
 }
 
+/**
+ * @param {import('dictionary').TermDictionaryEntry|import('dictionary').KanjiDictionaryEntry} dictionaryEntry
+ * @returns {number[]}
+ */
+function getFrequencyNumbers(dictionaryEntry) {
+    let previousDictionary;
+    const frequencies = [];
+    for (const {dictionary, frequency, displayValue} of dictionaryEntry.frequencies) {
+        if (dictionary === previousDictionary) {
+            continue;
+        }
+        previousDictionary = dictionary;
+
+        const frequencyWorking = displayValue !== null ? displayValue.match(/\d+/) : frequency;
```

Comment:
IMO this particular logic should still be changed to something more similar to what I mentioned before, since what's happening is actually a bit subtle.

The `Number(frequencyWorking)` conversion does one of two things:
1. Converts the value that is already a number (`frequency`) to a number, so a no-op.
2. Converts the regex match result to a number, but the way that it does this is a bit odd, and it only works due to the regex being formatted the way it is

   `frequencyWorking` will be a `RegExpMatchArray`, which is a special array with custom properties attached to it. The custom properties aren't relevant, but the fact that it's an array is. Passing this into the `Number` function will first convert the argument to a string. For arrays, this stringifies each element and concatenates them with `','`. As it turns out, the `RegExpMatchArray`'s length is `1` because the regex has no capture groups. but consider the following:

   `Number('1234'.match(/(\d+)/))` yields `NaN` because it is effectively doing `Number('1234,1234')`
   
   So what I think would be more clear type safety would be to do something similar to what I mentioned in a previous comment with the `getFrequencyNumber` function. The reason why it being in a separate function helps is because it allows you to just do `frequencies.push(getFrequencyNumber(the, arguments))`, which only takes one `.push` call and is a bit easier to do with a `return` in a function since it's nested two if's deep.


Additionally, I recommend either `parseFloat(v)` or `parseInt(v, 10)` due to the scope of usage being more explicitly limited to string conversions, and also because those are the more commonly used functions throughout the codebase.

Please enjoy this table of JavaScript madness:
| expression | result |
| --- | --- |
| `Number.parseInt('1234,1234', 10)` | `1234` |
| `Number.parseFloat('1234,1234')` | `1234` |
| `Number('1234,1234')` | `NaN` |

---

Now, one could argue that none of this matters because it works as it is and performance is irrelevant for this function, but I am pedantic.

Reply:
Well that's insane... 

I've changed this to use parseInt now. I quite like just being able to toss the dict info to `getFrequencyNumbers` and getting back the nice data instead of having to do any checks over the data in `getFrequencyHarmonic` or `getFrequencyAverage`. 

Using `continue` in the loop kindof does what `return` would do if I had split it out exactly like you said so that hopefully cleans up that well enough.